cmd/snap-seccomp/main.go

```diff
@@ -185,8 +189,7 @@ import (
 	"strings"
 	"syscall"
 
-	// FIXME: we want github.com/seccomp/libseccomp-golang but that will not work with trusty because libseccomp-golang checks for the seccomp version and errors if it find one < 2.2.0
-	"github.com/mvo5/libseccomp-golang"
+	"github.com/seccomp/libseccomp-golang"
```

Comment:
Unfortunately this change will break building on trusty. This is something we still want to support if at all possible. Is there a way to avoid this? does the forked repo need an update or something?

Reply:
@mvo5 this is intentional as explained in the commit message https://github.com/snapcore/snapd/pull/10808/commits/7bd6d7451e529a929c4b2836fc815069ea74ebdd

The last attempt to build snapd on trusty was 2.43.3~14.04 and it never got released. Thus snapd is no longer built on trusty at all. The snapd snap built on xenial, may still be used on trusty.

Reply:
Fwiw, I agree with Ian here, if we have some sort of catastrophic security issue and need SRU/ESM snapd back to trusty I would hate to be in a situation where we can't do it. It sounds like we need to involve the security here, either we can drop snapd in trusty from ESM (which is problematic as live-patch on trusty requires snapd) or we need to find a way to either keep working with a patched version of seccomp-golang or get an update of libseccomp 2.2 into trusty.

Reply:
Let's explore this a bit more, I created a https://github.com/mvo5/libseccomp-golang/tree/move-to-seccomp-golang-20210920 branch that moves to tip of libseccomp-golang and applies our patches on top, I started a spread run to see if 14.04 still builds with that.

Reply:
Fwiw, if we keep using the fork we should move it to "snapcore" out of my personal repo for sure.

Reply:
@anonymouse64 @mvo5 ESM policies prevent from landing new snapd releases. If you need to build and release new snapd to trusty it would have to be a patch on top of 2.38~14.04

Also, we must continue to test xenial built snap on trusty host. Is this being done?

Reply:
@anonymouse64 @mvo5 trusty SRU is closed; only ESM security remains open for Trusty, which prohibits accepting anything but High+ targeted security fixes only.

Reply:
I disagree that upstream libseecomp is used for distro package builds, and not used for re-execed snapd on those hosts.

Surely, we should be patching in to use mvo5's fork on trusty only these days? which is only one combination of them all, instead of forcing to maintain & use a fork on all builds.

Reply:
@xnox when I said "SRU" I meant SRU via ESM for trusty. The page specifically about ESM on trusty specifically says of snapd under "exclusions":
```
Various packages related to snapd and working with snap packages (not included on installation media). snapd itself is supported to the extent of ensuring livepatch and cvescan remain functional with CVE fixes only for the code representing the operational aspects of livepatch and cvescan functionality.
```

Though actually reading it, it says cvescan too, which is not a classic snap which is a bit troubling since we ignored other security vulnerabilities for trusty since we thought trusty only mattered for livepatch only which is a classic snap so security is kinda nil there. Hmmmmm

Also @xnox we have historically been able to get new version numbers of snapd SRU'd via ESM to xenial, so I think the real answer to this discussion is to discuss options with the security team.

And I totally agree that the best case scenario would be to use mvo's patch only where we need to on trusty and use upstream everywhere else but this is more difficult to maintain for us.

Reply:
> And I totally agree that the best case scenario would be to use mvo's patch only where we need to on trusty and use upstream everywhere else but this is more difficult to maintain for us.

huh, this is already maintained in a complicated way in current snapd code, where upstream seccomp is used in fedora/debian builds, but not in any others. With a dynamic monkey patch. My proposal is to reverse the monkey patch and move it to trusty-only build path.

Reply:
I like the idea of using the fork only on trusty fwiw, I hope now that we have go.mod with some "replace" trickery in there this because actually not too messy. In any case, the update fork branch seems to build so that is (now) also an option and regardless we should switch to it.